stp_zmq/zstack.py

```diff
@@ -753,17 +776,21 @@ def transmitThroughListener(self, msg, ident) -> Tuple[bool, Optional[str]]:
                          format(self, msg, ident))
             self.listener.send_multipart([ident, msg], flags=zmq.NOBLOCK)
         except zmq.Again:
-            return False, None
-        except InvalidMessageExceedingSizeException as ex:
-            err_str = '{}Cannot transmit message. Error {}'.format(CONNECTION_PREFIX, ex)
+            raise EAgainError(msg, ident, prefix=prefix)
```

Comment:
Should we raise exception here?

Reply:
It depends... Actually for current synchronous non-blocking usage of zmq we need to implement logic to repeat transmit socket write operations for zmq.EAgain errors. For now there is no such logic and I wasn't going to do that in scope of current PR. Thus, the exception is passed upper. And it's not only error that we pass upper where all such errors are just logged with warnings: we lose (and ignore that) all out messages even if it can bee fixed in some cases.